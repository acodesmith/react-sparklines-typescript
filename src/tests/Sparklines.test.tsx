import React from "react";
import { render } from "@testing-library/react";
import {
  Sparklines,
  SparklinesLine,
  SparklinesBars,
  SparklinesCurve,
  SparklinesNormalBand,
  SparklinesReferenceLine,
  SparklinesSpots,
  SparklinesText,
} from "../Sparklines";
import "@testing-library/jest-dom/extend-expect";

describe("Basic Sparkline Component Rendering", () => {
  it("should return null", () => {
    const component = (
      <Sparklines data={[]}>
        <SparklinesLine />
      </Sparklines>
    );
    const { container } = render(component);
    expect(container.firstChild).toEqual(null);
  });

  it("should render a simple sparkline with sparkline lines", () => {
    const component = (
      <Sparklines data={[1, 2]}>
        <SparklinesLine />
      </Sparklines>
    );
    const { container } = render(component);
    expect(container).toMatchSnapshot();
  });

  it("should render a simple sparkline with sparkline lines and set width/height", () => {
    const component = (
      <Sparklines svgWidth={100} svgHeight={100} data={[1, 2]}>
        <SparklinesLine />
      </Sparklines>
    );
    const { container } = render(component);
    expect(container).toMatchSnapshot();
  });

  it("should render all the sparkline pieces together", () => {
    const component = (
      <Sparklines
        svgWidth={100}
        svgHeight={100}
        data={[1, 2, 3, 4, 5, 6, 7, 8, 9]}
      >
        <SparklinesLine />
        <SparklinesBars />
        <SparklinesCurve />
        <SparklinesNormalBand />
        <SparklinesReferenceLine />
        <SparklinesSpots />
        <SparklinesText text={"Hello!"} />
      </Sparklines>
    );
    const { container } = render(component);
    expect(container).toMatchSnapshot();
  });
});
