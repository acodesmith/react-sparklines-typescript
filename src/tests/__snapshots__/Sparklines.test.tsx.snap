// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Sparkline Component Rendering should render a simple sparkline with sparkline lines 1`] = `
<div>
  <svg
    preserveAspectRatio="none"
    viewBox="0 0 240 60"
  >
    <g>
      <circle
        cx="2"
        cy="58"
        r="2"
        style="stroke: none; stroke-width: 0; fill-opacity: 0.1; fill: slategray; pointer-events: auto;"
      />
      <circle
        cx="238"
        cy="2"
        r="2"
        style="stroke: none; stroke-width: 0; fill-opacity: 0.1; fill: slategray; pointer-events: auto;"
      />
      <polyline
        points="2 58 238 2 238 58 2 58 2 58"
        style="stroke: none; stroke-width: 0; fill-opacity: 0.1; fill: slategray; pointer-events: auto;"
      />
      <polyline
        points="2 58 238 2"
        style="stroke: slategray; stroke-width: 1; stroke-linejoin: round; stroke-linecap: round; fill: none;"
      />
    </g>
  </svg>
</div>
`;

exports[`Basic Sparkline Component Rendering should render a simple sparkline with sparkline lines and set width/height 1`] = `
<div>
  <svg
    height="100"
    preserveAspectRatio="none"
    viewBox="0 0 240 60"
    width="100"
  >
    <g>
      <circle
        cx="2"
        cy="58"
        r="2"
        style="stroke: none; stroke-width: 0; fill-opacity: 0.1; fill: slategray; pointer-events: auto;"
      />
      <circle
        cx="238"
        cy="2"
        r="2"
        style="stroke: none; stroke-width: 0; fill-opacity: 0.1; fill: slategray; pointer-events: auto;"
      />
      <polyline
        points="2 58 238 2 238 58 2 58 2 58"
        style="stroke: none; stroke-width: 0; fill-opacity: 0.1; fill: slategray; pointer-events: auto;"
      />
      <polyline
        points="2 58 238 2"
        style="stroke: slategray; stroke-width: 1; stroke-linejoin: round; stroke-linecap: round; fill: none;"
      />
    </g>
  </svg>
</div>
`;

exports[`Basic Sparkline Component Rendering should render all the sparkline pieces together 1`] = `
<div>
  <svg
    height="100"
    preserveAspectRatio="none"
    viewBox="0 0 240 60"
    width="100"
  >
    <g>
      <circle
        cx="2"
        cy="58"
        r="2"
        style="stroke: none; stroke-width: 0; fill-opacity: 0.1; fill: slategray; pointer-events: auto;"
      />
      <circle
        cx="31.5"
        cy="51"
        r="2"
        style="stroke: none; stroke-width: 0; fill-opacity: 0.1; fill: slategray; pointer-events: auto;"
      />
      <circle
        cx="61"
        cy="44"
        r="2"
        style="stroke: none; stroke-width: 0; fill-opacity: 0.1; fill: slategray; pointer-events: auto;"
      />
      <circle
        cx="90.5"
        cy="37"
        r="2"
        style="stroke: none; stroke-width: 0; fill-opacity: 0.1; fill: slategray; pointer-events: auto;"
      />
      <circle
        cx="120"
        cy="30"
        r="2"
        style="stroke: none; stroke-width: 0; fill-opacity: 0.1; fill: slategray; pointer-events: auto;"
      />
      <circle
        cx="149.5"
        cy="23"
        r="2"
        style="stroke: none; stroke-width: 0; fill-opacity: 0.1; fill: slategray; pointer-events: auto;"
      />
      <circle
        cx="179"
        cy="16"
        r="2"
        style="stroke: none; stroke-width: 0; fill-opacity: 0.1; fill: slategray; pointer-events: auto;"
      />
      <circle
        cx="208.5"
        cy="9"
        r="2"
        style="stroke: none; stroke-width: 0; fill-opacity: 0.1; fill: slategray; pointer-events: auto;"
      />
      <circle
        cx="238"
        cy="2"
        r="2"
        style="stroke: none; stroke-width: 0; fill-opacity: 0.1; fill: slategray; pointer-events: auto;"
      />
      <polyline
        points="2 58 31.5 51 61 44 90.5 37 120 30 149.5 23 179 16 208.5 9 238 2 238 58 2 58 2 58"
        style="stroke: none; stroke-width: 0; fill-opacity: 0.1; fill: slategray; pointer-events: auto;"
      />
      <polyline
        points="2 58 31.5 51 61 44 90.5 37 120 30 149.5 23 179 16 208.5 9 238 2"
        style="stroke: slategray; stroke-width: 1; stroke-linejoin: round; stroke-linecap: round; fill: none;"
      />
    </g>
    <g
      transform="scale(1,-1)"
    >
      <rect
        height="2"
        style="fill: slategray;"
        width="25.5"
        x="-10.75"
        y="-60"
      />
      <rect
        height="9"
        style="fill: slategray;"
        width="25.5"
        x="18.75"
        y="-60"
      />
      <rect
        height="16"
        style="fill: slategray;"
        width="25.5"
        x="48.25"
        y="-60"
      />
      <rect
        height="23"
        style="fill: slategray;"
        width="25.5"
        x="77.75"
        y="-60"
      />
      <rect
        height="30"
        style="fill: slategray;"
        width="25.5"
        x="107.25"
        y="-60"
      />
      <rect
        height="37"
        style="fill: slategray;"
        width="25.5"
        x="136.75"
        y="-60"
      />
      <rect
        height="44"
        style="fill: slategray;"
        width="25.5"
        x="166.25"
        y="-60"
      />
      <rect
        height="51"
        style="fill: slategray;"
        width="25.5"
        x="195.75"
        y="-60"
      />
      <rect
        height="58"
        style="fill: slategray;"
        width="25.5"
        x="225.25"
        y="-60"
      />
    </g>
    <g>
      <path
        d="M2 58 C 9.375 58 24.125 51 31.5 51 C 38.875 51 53.625 44 61 44 C 68.375 44 83.125 37 90.5 37 C 97.875 37 112.625 30 120 30 C 127.375 30 142.125 23 149.5 23 C 156.875 23 171.625 16 179 16 C 186.375 16 201.125 9 208.5 9 C 215.875 9 230.625 2 238 2 L238 58 2 58 2 58"
        style="stroke: none; stroke-width: 0; fill-opacity: 0.1; fill: slategray;"
      />
      <path
        d="M2 58 C 9.375 58 24.125 51 31.5 51 C 38.875 51 53.625 44 61 44 C 68.375 44 83.125 37 90.5 37 C 97.875 37 112.625 30 120 30 C 127.375 30 142.125 23 149.5 23 C 156.875 23 171.625 16 179 16 C 186.375 16 201.125 9 208.5 9 C 215.875 9 230.625 2 238 2"
        style="stroke: slategray; stroke-width: 1; stroke-linejoin: round; stroke-linecap: round; fill: none;"
      />
    </g>
    <rect
      height="NaN"
      style="fill: red; fill-opacity: 0.1;"
      width="236"
      x="2"
      y="13.926077717698721"
    />
    <line
      style="stroke: red; stroke-opacity: 0.75; stroke-dasharray: 2, 2;"
      x1="2"
      x2="238"
      y1="32"
      y2="32"
    />
    <g>
      <circle
        cx="238"
        cy="2"
      />
    </g>
    <g>
      <text
        font-family="Verdana"
        font-size="10"
      >
        Hello!
      </text>
    </g>
  </svg>
</div>
`;
